Mathlib/MeasureTheory/Measure/LevyProkhorovMetric.lean

```diff
@@ -387,19 +435,220 @@ lemma continuous_levyProkhorov_to_probabilityMeasure :
+  by_cases easy : P.toMeasure G < ε
+  · exact eventually_of_forall (fun _ ↦ lt_of_lt_of_le easy le_add_self)
+  simp only [not_lt] at easy
+  have aux : P.toMeasure G - ε < liminf (fun Q ↦ Q.toMeasure G) (𝓝 P) := by
+    apply lt_of_lt_of_le (ENNReal.sub_lt_self (measure_lt_top _ _).ne _ _)
+        <| ProbabilityMeasure.le_liminf_measure_open_of_tendsto tendsto_id G_open
+    · exact (lt_of_lt_of_le ε_pos easy).ne.symm
+    · exact ε_pos.ne.symm
+  filter_upwards [gt_mem_sets_of_limsInf_gt (α := ℝ≥0∞) isBounded_ge_of_bot
+      (show P.toMeasure G - ε < limsInf ((𝓝 P).map (fun Q ↦ Q.toMeasure G)) from aux)] with Q hQ
+  simp only [preimage_setOf_eq, mem_setOf_eq] at hQ
+  convert ENNReal.add_lt_add_right ε_ne_top hQ
+  exact (tsub_add_cancel_of_le easy).symm
+
+lemma ProbabilityMeasure.continuous_toLevyProkhorov :
+    Continuous (ProbabilityMeasure.toLevyProkhorov (Ω := Ω)) := by
+
+  -- We will check continuity of `id : ProbabilityMeasure Ω → LevyProkhorov (ProbabilityMeasure Ω)`
+  -- at each point `P : ProbabilityMeasure Ω`.
+  rw [continuous_iff_continuousAt]
+  intro P
+
+  -- To check continuity, fix `ε > 0`. To leave some wiggle room, be ready to use `ε/3 > 0` instead.
+  rw [continuousAt_iff']
+  intro ε ε_pos
+  have third_ε_pos : 0 < ε / 3 := by linarith
+  have third_ε_pos' : 0 < ENNReal.ofReal (ε / 3) := ofReal_pos.mpr third_ε_pos
+
+  -- First use separability to choose a countable partition of `Ω` into measurable
+  -- subsets `Es n ⊆ Ω` of small diamater, `diam (Es n) < ε/3`.
+  obtain ⟨Es, Es_mble, Es_bdd, Es_diam, Es_cover, Es_disjoint⟩ :=
+    SeparableSpace.exists_measurable_partition_diam_le Ω third_ε_pos
+
+  -- Instead of the whole space `Ω = ⋃ n ∈ ℕ, Es n`, focus on a large but finite
+  -- union `⋃ n < N, Es n`, chosen in such a way that the complement has small `P`-mass,
+  -- `P (⋃ n < N, Es n)ᶜ < ε/3`.
+  have exhaust :=
+    @tendsto_measure_biUnion_Ici_zero_of_pairwise_disjoint Ω _ P.toMeasure _ Es Es_mble Es_disjoint
+  simp only [tendsto_atTop_nhds, Function.comp_apply] at exhaust
+  obtain ⟨N, hN⟩ := exhaust (Iio (ENNReal.ofReal (ε / 3))) third_ε_pos' isOpen_Iio
+  specialize hN N le_rfl
+
+  -- With the finite `N` fixed above, consider the finite collection of open sets of the form
+  -- `Gs J = thickening (ε/3) (⋃ j ∈ J, Es j)`, where `J ⊆ {0, 1, ..., N-1}`.
+  have Js_finite : Set.Finite {J | J ⊆ Iio N} := Finite.finite_subsets <| finite_Iio N
+  set Gs := (fun (J : Set ℕ) ↦ thickening (ε/3) (⋃ j ∈ J, Es j)) '' {J | J ⊆ Iio N}
+  have Gs_open : ∀ (J : Set ℕ), IsOpen (thickening (ε/3) (⋃ j ∈ J, Es j)) :=
+    fun J ↦ isOpen_thickening
+
+  -- Any open set `G ⊆ Ω` determines a neighborhood of `P` consisting of those `Q` that
+  -- satisfy `P G < Q G + ε/3`.
+  have mem_nhds_P :
+      ∀ G, IsOpen G → ({Q | P.toMeasure G < Q.toMeasure G + ENNReal.ofReal (ε/3)} ∈ 𝓝 P) :=
+    fun G G_open ↦ P.toMeasure_add_pos_gt_mem_nhds G_open third_ε_pos' ofReal_ne_top
```

Comment:
```suggestion
  have mem_nhds_P (G : Set Ω) (G_open : IsOpen G) : 
      {Q | P.toMeasure G < Q.toMeasure G + ENNReal.ofReal (ε/3)} ∈ 𝓝 P :=
    P.toMeasure_add_pos_gt_mem_nhds G_open third_ε_pos' ofReal_ne_top
```
You can move variables to the left of the colon in `have` statements.

Reply:
Ah, I did not realize that `have` allows such syntax. Thanks!